launcher/minecraft/launch/DirectJavaLaunch.cpp

```diff
@@ -50,7 +55,32 @@ void DirectJavaLaunch::executeTask()
 
     auto javaPath = FS::ResolveExecutable(instance->settings()->get("JavaPath").toString());
 
-    m_process.setProcessEnvironment(instance->createEnvironment());
+    auto env = instance->createEnvironment();
+#ifdef Q_OS_LINUX
+    if (instance->settings()->get("EnableMangoHud").toBool())
+    {
+        auto preload = env.value("LD_PRELOAD", "") + ":libMangoHud_dlsym.so:libMangoHud.so";
+        auto lib_path = env.value("LD_LIBRARY_PATH", "") +  ":/usr/local/lib64/mangohud/:/usr/local/lib/mangohud/:/usr/lib64/mangohud/:/usr/lib/mangohud/";
```

Comment:
Do we need to touch LD_LIBRARY_PATH? From what I can tell, mangohud is generally installed at `/usr/lib{,64}/libMangoHud{,_dlsym}.so`. If it isn't, the user should deal with this manually (e.g. setting `GAME_LIBRARY_PATH` as an env-var for PolyMC, see ![JavaUtils::CleanEnviroment](https://github.com/PolyMC/PolyMC/blob/e22d54abd35560213377bb41f2bc902f67c8de64/launcher/java/JavaUtils.cpp#L86-L111))

Reply:
Sure, but you would expect that enabling the MangoHud option would automatically deal with that since you explicitly said you want to use MangoHud.

Simpling throwing libMangoHud into LD_PRELOAD does nothing since `/usr/{LIB}/mangohud` is not on the default search path.

Since writing this I've found out that `ld.so` can parse dynamic string tokens so instead of having that long mess of a lib path string we can simply do `":/usr/local/$LIB/mangohud/:/usr/$LIB/mangohud/` and it will correctly deal with it.

Reply:
Is this the packaging on Fedora? On Arch mangohud is just thrown into `/usr/lib` (no sub-directory)

Reply:
Yeah that is how its packaged for Fedora, I had been unaware that there are different setups for this.

Looking a bit deeper, both configurations are supported by upstream
https://github.com/flightlessmango/MangoHud/blob/a8a0a245e69fbbca5263d2436fd1c04289375498/src/meson.build#L4-L7

Reply:
Okay so `":/usr/local/$LIB/:/usr/local/$LIB/mangohud/:/usr/$LIB/mangohud/` should be best.

Reply:
> Okay, so `":/usr/local/$LIB/:/usr/local/$LIB/mangohud/:/usr/$LIB/mangohud/` should be best if we really want to `local` to LD_LIBRARY_PATH. (Keep in mind that this also affects java and all natives)

isn't `/usr/local/$LIB` already in the default search path?

Reply:
> isn't `/usr/local/$LIB` already in the default search path?

Nope. Most distributions (userbase-wise) don't include it in LD_LIBRARY_PATH. Notably Ubuntu (Debian?) and Arch

Reply:
looking further into it this isn't the case on Red Hat distros either, I've just been cmake was just doing rpath fuckery which made me believe it was.